### ISSUE TITLE
Docker desktop 4.36.0 fails to start with "more than one sd* disk in /sys/block"

### ISSUE THREAD

**Issue by chris-riddell35** [docker/for-win#14453]
### Description

Unable to start docker desktop version 4.36.0; I am consistently seeing the following error:

`running engine: waiting for the Docker API: engine linux/wsl failed to run: starting WSL engine: error spotted in wslbootstrap log: "[2024-11-28T16:04:44.799442528Z][wsl-bootstrap][F] more than one sd* disk in /sys/block with wwid ending by 57de6aa93848de02790adcb4: /dev/sdb,/dev/sdd"`

### Reproduce

Start docker desktop

### Expected behavior

Docker desktop should start successfully

### docker version

```bash
Client:
 Version:           27.3.1
 API version:       1.47
 Go version:        go1.22.7
 Git commit:        ce12230
 Built:             Fri Sep 20 11:42:27 2024
 OS/Arch:           windows/amd64
 Context:           desktop-linux
```

### docker info

```bash
Client:
 Version:    27.3.1
 Context:    desktop-linux
 Debug Mode: false
 Plugins:
  ai: Ask Gordon - Docker Agent (Docker Inc.)
    Version:  v0.1.0
    Path:     C:\Program Files\Docker\cli-plugins\docker-ai.exe
  buildx: Docker Buildx (Docker Inc.)
    Version:  v0.18.0-desktop.2
    Path:     C:\Program Files\Docker\cli-plugins\docker-buildx.exe
  compose: Docker Compose (Docker Inc.)
    Version:  v2.30.3-desktop.1
    Path:     C:\Program Files\Docker\cli-plugins\docker-compose.exe
  debug: Get a shell into any image or container (Docker Inc.)
    Version:  0.0.37
    Path:     C:\Program Files\Docker\cli-plugins\docker-debug.exe
  desktop: Docker Desktop commands (Alpha) (Docker Inc.)
    Version:  v0.0.15
    Path:     C:\Program Files\Docker\cli-plugins\docker-desktop.exe
  dev: Docker Dev Environments (Docker Inc.)
    Version:  v0.1.2
    Path:     C:\Program Files\Docker\cli-plugins\docker-dev.exe
  extension: Manages Docker extensions (Docker Inc.)
    Version:  v0.2.27
    Path:     C:\Program Files\Docker\cli-plugins\docker-extension.exe
  feedback: Provide feedback, right in your terminal! (Docker Inc.)
    Version:  v1.0.5
    Path:     C:\Program Files\Docker\cli-plugins\docker-feedback.exe
  init: Creates Docker-related starter files for your project (Docker Inc.)
    Version:  v1.4.0
    Path:     C:\Program Files\Docker\cli-plugins\docker-init.exe
  sbom: View the packaged-based Software Bill Of Materials (SBOM) for an image (Anchore Inc.)
    Version:  0.6.0
    Path:     C:\Program Files\Docker\cli-plugins\docker-sbom.exe
  scout: Docker Scout (Docker Inc.)
    Version:  v1.15.0
    Path:     C:\Program Files\Docker\cli-plugins\docker-scout.exe

Server:
ERROR: Get "http://%2F%2F.%2Fpipe%2FdockerDesktopLinuxEngine/v1.47/info": context canceled
errors pretty printing info
```

### Diagnostics ID

1AA58D3A-8AF2-493C-AEAF-41A589D2DF25/20241128162112

### Additional Info

WSL version info:
```
wsl --version
WSL version: 2.3.26.0
Kernel version: 5.15.167.4-1
WSLg version: 1.0.65
MSRDC version: 1.2.5620
Direct3D version: 1.611.1-81528511
DXCore version: 10.0.26100.1-240331-1435.ge-release
Windows version: 10.0.22631.4460
```

**Comment by MarcelBolten** [comment]
Hi,

I'm also observing the same error:

`running engine: waiting for the Docker API: engine linux/wsl failed to run: starting WSL engine: error spotted in wslbootstrap log: "[2024-11-28T19:48:55.345483430Z][wsl-bootstrap][F] more than one sd* disk in /sys/block with wwid ending by 57de6aa93848de02790adcb4: /dev/sdc,/dev/sde"`

### docker version
```
Client:
 Version:           27.3.1
 API version:       1.47
 Go version:        go1.22.7
 Git commit:        ce12230
 Built:             Fri Sep 20 11:42:27 2024
 OS/Arch:           windows/amd64
 Context:           desktop-linux
error during connect: Get "http://%2F%2F.%2Fpipe%2FdockerDesktopLinuxEngine/v1.47/version": open //./pipe/dockerDesktopLinuxEngine: The system cannot find the file specified.
```

### docker info
```
Client:
 Version:    27.3.1
 Context:    desktop-linux
 Debug Mode: false
 Plugins:
  ai: Ask Gordon - Docker Agent (Docker Inc.)
    Version:  v0.1.0
    Path:     C:\Program Files\Docker\cli-plugins\docker-ai.exe
  buildx: Docker Buildx (Docker Inc.)
    Version:  v0.18.0-desktop.2
    Path:     C:\Program Files\Docker\cli-plugins\docker-buildx.exe
  compose: Docker Compose (Docker Inc.)
    Version:  v2.30.3-desktop.1
    Path:     C:\Program Files\Docker\cli-plugins\docker-compose.exe
  debug: Get a shell into any image or container (Docker Inc.)
    Version:  0.0.37
    Path:     C:\Program Files\Docker\cli-plugins\docker-debug.exe
  desktop: Docker Desktop commands (Alpha) (Docker Inc.)
    Version:  v0.0.15
    Path:     C:\Program Files\Docker\cli-plugins\docker-desktop.exe
  dev: Docker Dev Environments (Docker Inc.)
    Version:  v0.1.2
    Path:     C:\Program Files\Docker\cli-plugins\docker-dev.exe
  extension: Manages Docker extensions (Docker Inc.)
    Version:  v0.2.27
    Path:     C:\Program Files\Docker\cli-plugins\docker-extension.exe
  feedback: Provide feedback, right in your terminal! (Docker Inc.)
    Version:  v1.0.5
    Path:     C:\Program Files\Docker\cli-plugins\docker-feedback.exe
  init: Creates Docker-related starter files for your project (Docker Inc.)
    Version:  v1.4.0
    Path:     C:\Program Files\Docker\cli-plugins\docker-init.exe
  sbom: View the packaged-based Software Bill Of Materials (SBOM) for an image (Anchore Inc.)
    Version:  0.6.0
    Path:     C:\Program Files\Docker\cli-plugins\docker-sbom.exe
  scout: Docker Scout (Docker Inc.)
    Version:  v1.15.0
    Path:     C:\Program Files\Docker\cli-plugins\docker-scout.exe

Server:
ERROR: error during connect: Get "http://%2F%2F.%2Fpipe%2FdockerDesktopLinuxEngine/v1.47/info": open //./pipe/dockerDesktopLinuxEngine: The system cannot find the file specified.
errors pretty printing info
```

### Diagnostics ID

ECCA2C28-89D7-4B8B-93C2-DBB853C4B6E2/20241128194210

### Additional Info

WSL version info:
```
WSL version: 2.3.26.0
Kernel version: 5.15.167.4-1
WSLg version: 1.0.65
MSRDC version: 1.2.5620
Direct3D version: 1.611.1-81528511
DXCore version: 10.0.26100.1-240331-1435.ge-release
Windows version: 10.0.22631.4460
```

**Comment by beegotsy** [comment]
For those who need to get Docker working; rollbacking to [4.35.1](https://desktop.docker.com/win/main/amd64/173168/Docker%20Desktop%20Installer.exe) ([release page](https://docs.docker.com/desktop/release-notes/)) works:
```
# install pending WSL updates with
wsl.exe --update
# install the previous version of Docker for Windows with:
.\<installer path>.exe install --disable-version-check
```
You will need to rebuild all images and containers, or at least that's what I did.
For me, the WSL update changed the error from a generic one to `more than one sd* disk in /sys/block with wwid ending by ...`.

This seems a fix to get 4.36.0 working: https://github.com/docker/for-win/issues/14447#issuecomment-2504140336

**Comment by griesi007** [comment]
same issue for me https://github.com/docker/for-win/issues/14445

**Comment by rupasix** [comment]
I have same issue, screenshot from error: 

![Image](https://github.com/user-attachments/assets/06657dad-2332-4d6c-9b82-045e40f734bd)

After that i checked with command "lsblk" if there is sde block and there was none:
![Image](https://github.com/user-attachments/assets/bc60672f-175e-4b72-870f-37a77a60462f)

**Comment by lerouxm** [comment]
Same issue here with an up to date wsl.
I can confirm that rolling back to previous 3.5.1 fixed the problem for me for now.

**Comment by dev-mich** [comment]
I had the same issue, but completely uninstalling and reinstalling resolved it for me.

**Comment by bastian-kurz** [comment]
I had the same issue. I solved it by going to 'Troubleshoot,' then clicking on 'Clean/Purge Data,' and selecting 'Hyper-V & WSL2.

![Image](https://github.com/user-attachments/assets/3a46afcf-b0ef-48a3-aafa-8fcb5e75a056)

after that Docker Engine starts again.

**Comment by andrea-reale** [comment]
Hello everyone,

thanks for reporting the problem. Could you please share the output of `wsl --list`?

From the diagnostics kindly shared by @chris-riddell35 , it looks like WSL2 is in a state where it believes there are two disks with the same identifiers, which makes Docker Desktop boot fail.

Could you please confirm if shutting down WSL after quitting Docker Desktop (`wsl.exe --shutdown`) helps fixing the issue or not? (Alternative a Windows restart should also be equivalent)

**Comment by chris-riddell35** [comment]
Hi @andrea-reale ,

I've done several Windows restarts, with the same result each time.

Here's the output of `wsl --list`:

```
wsl --list -v
  NAME              STATE           VERSION
* Ubuntu-20.04      Running         2
  Ubuntu            Stopped         1
  Ubuntu-24.04      Stopped         2
  docker-desktop    Stopped         2
```

**Comment by andrea-reale** [comment]
Hey @chris-riddell35 ,

thanks for your quick response and for confirming. I am having a hard time to try and reproduce the issue.
However, I will be shortly sharing a development build of Docker Desktop with a tentative fix. 

I'd sincerely appreciate if you could help us by trying it out and confirm if it addresses the problem or not. If it does, I'll try to get it included in the upcoming 4.37 release.

**Comment by andrea-reale** [comment]
@chris-riddell35, all: as promised here's a **development build** of Docker Desktop 4.36 that should **includes a fix for the issue**: [installer for windows x86](https://desktop-stage.docker.com/win/main/amd64/176956/Docker%20Desktop%20Installer.exe).  

Please, let me know if that helps with the `more than one sd* disk` error discussed in this ticket, and feel free to provide any further feedback (make sure to share diagnostics in case you stumble into different errors).  

**Note** that being a development build, the installer is unsigned and you'll need to tell Windows you trust it, before it will let you run it. You can do so by right-clicking on the file  -> `Properties` -> `Unblock`.

**Comment by thomaswiradikusuma** [comment]
I had exactly this error as well, right after upgrading. It disappears after few times open-crash-close-open-crash-close-open :)

**Comment by Cristy94** [comment]
I have the same issue with 4.36.0, it suddenly stopped working.

```
running engine: waiting for the Docker API: engine linux/wsl failed to run: starting WSL engine: error spotted in wslbootstrap log: "[2024-11-29T15:17:21.165474510Z][wsl-bootstrap][F] more than one sd* disk in /sys/block with wwid ending by 57de6aa93848de02790adcb4: /dev/sdc,/dev/sdd"
```

```
wsl --list
Windows Subsystem for Linux Distributions:
docker-desktop (Default)
```

**Comment by Cristy94** [comment]
> I had the same issue. I solved it by going to 'Troubleshoot,' then clicking on 'Clean/Purge Data,' and selecting 'Hyper-V & WSL2.
> 
> ![Image](https://github.com/user-attachments/assets/3a46afcf-b0ef-48a3-aafa-8fcb5e75a056)
> 
> after that Docker Engine starts again.

Where did you find "Troubleshoot"?

Nevermind, it's here:

![Image](https://github.com/user-attachments/assets/4521ac92-f6d2-4462-9cb7-ef8fde65a0eb)

**EDIT:**

Purging the data works (note that you will lose all containers).

**Comment by salvorapi** [comment]
> [@chris-riddell35](https://github.com/chris-riddell35), all: as promised here's a **development build** of Docker Desktop 4.36 that should **includes a fix for the issue**: [installer for windows x86](https://desktop-stage.docker.com/win/main/amd64/176956/Docker%20Desktop%20Installer.exe).
> 
> Please, let me know if that helps with the `more than one sd* disk` error discussed in this ticket, and feel free to provide any further feedback (make sure to share diagnostics in case you stumble into different errors).
> 
> **Note** that being a development build, the installer is unsigned and you'll need to tell Windows you trust it, before it will let you run it. You can do so by right-clicking on the file -> `Properties` -> `Unblock`.

it work 👍

**Comment by chris-riddell35** [comment]
@andrea-reale after installing and restarting, docker desktop is up and running but something is wrong with the wsl2 integration. I see this error message:

![Image](https://github.com/user-attachments/assets/681a154c-bebf-479c-abf2-19d606b4de3f)


Diagnostics ID:
1AA58D3A-8AF2-493C-AEAF-41A589D2DF25/20241129155916

If I run docker from powershell, it's working. I typically use wsl2 and I have all my configurations there so this is still inconvenient, but appears to be workable. Let me know if I can gather more information about the new error.

**Comment by chris-riddell35** [comment]
@andrea-reale I disabled the WSL integration, re-enabled and restarted Windows and this seems to be working now. I lost all my images when I did that, but I think the system is ok now.

I had also had two wsl2 distros active there, and I disabled one of them. I guess that may have helped fix this issue.

![Image](https://github.com/user-attachments/assets/a30fa0f7-8a03-4353-b34a-8c6e013002ad)

**Comment by andrea-reale** [comment]
Hey @chris-riddell35 , thanks for reporting back your experience.

Glad to hear that [the dev build](https://github.com/docker/for-win/issues/14453#issuecomment-2507992110) fixed at least the startup issue. And I am sorry about the troubles with the Ubuntu integration. It looks like the previous integration moved to an unhealthy state after the upgrade: glad that disabling it and enabling it again mitigated the problem.

I'll look more into the diagnostics you provided to understand more about why the integration issue started to begin with.

**Comment by SparkInChief** [comment]
@andrea-reale — The dev build resolved the issue for me as well (same error message immediately after upgrading to 4.36.0). 

I did not have to reboot (or purge anything), but I did lose all locally cached images and had to rebuild all images and restart containers.

Here's output of `wsl --version` in case it helps determine what systems the dev build works for (and why it doesn't work on others):

```
WSL version: 2.3.26.0
Kernel version: 5.15.167.4-1
WSLg version: 1.0.65
MSRDC version: 1.2.5620
Direct3D version: 1.611.1-81528511
DXCore version: 10.0.26100.1-240331-1435.ge-release
Windows version: 10.0.22631.4541
```

**Comment by neonima1995** [comment]
Removing Ubuntu from Windows helped me.![Image](https://github.com/user-attachments/assets/a758ba7d-6fd0-4d08-9d08-016d5746867e)![Image](https://github.com/user-attachments/assets/d77b388b-ad22-4d53-b899-cb30e12262b3)

**Comment by queensjacket** [comment]
God damn does this not get tested before being released...

**Comment by theswissguy** [comment]
'Troubleshoot,' then clicking on 'Clean/Purge Data,' helped in my case. Docker Desktop starts, but in won't let my sign in. I'm been redirected to https://app.docker.com/auth/ and back to Docker Desktop, but all I get is

![Image](https://github.com/user-attachments/assets/11d7c921-1de9-4e88-9f04-b8f88b2d0e46)

**Comment by WesWedding** [comment]
> [@chris-riddell35](https://github.com/chris-riddell35), all: as promised here's a **development build** of Docker Desktop 4.36 that should **includes a fix for the issue**: [installer for windows x86](https://desktop-stage.docker.com/win/main/amd64/176956/Docker%20Desktop%20Installer.exe).
> 
> Please, let me know if that helps with the `more than one sd* disk` error discussed in this ticket, and feel free to provide any further feedback (make sure to share diagnostics in case you stumble into different errors).
> 
> **Note** that being a development build, the installer is unsigned and you'll need to tell Windows you trust it, before it will let you run it. You can do so by right-clicking on the file -> `Properties` -> `Unblock`.

This development package has addressed the issue for me.  I did not attempt to manually purge cached data yet, as mentioned in an earlier post by another user upthread.

**Comment by xmontero** [comment]
Same problem.

Solutions so far?
* It seems I could uninstall and re-install an older version. But it seems to loose images and containers. I can't afford that loose.
* Development build.

Question: a) "Dev build" means a re-installation of a new version? b) Will I loose any data by doing so? I have production data that I can't loose.

**Comment by xmontero** [comment]
> [@chris-riddell35](https://github.com/chris-riddell35), all: as promised here's a **development build** of Docker Desktop 4.36 that should **includes a fix for the issue**: [installer for windows x86](https://desktop-stage.docker.com/win/main/amd64/176956/Docker%20Desktop%20Installer.exe).

Is there any risk of loosing my images, containers, networks and/or volumes by running the installer?

**Comment by xmontero** [comment]
Hey, hey hey hey hey!!!

The installer **REMOVED** my docker-desktop-data volume!!!!

This means I lost **EVERYTHING**?!?!?!?!?!?!?

How can this happen without notice!? At least an alert or if you want at least don't using it but "leaving it alone"!!! But... removed **ALL MY DATA** !?!?!?

**Comment by xmontero** [comment]
Here's the complete sequence I did:

1. Downloaded the dev-version
2. Run the installer
3. After the installer run, I run docker desktop
4. A black window with many logs appeared and at some point Avast was persistently complaining. I thought as it was a dev version and non-signed maybe that was not good for avast so I disabled avast.
5. Finally the Desktop Docker GUI appeared **and I could see the images, containers, volumes, etc. there inside** so it seemed I did not loose anything.
6. I rebooted the PC to check all was in place.
7. After rebooting the PC, no images, no containers, no anything...
8. Doing `wsl --list` I see `Ubuntu-20.04` and `docker-desktop` but no `docker-desktop-data`.

May I recover the lost `docker-desktop-data` from somewhere? I'm happy to copy everything manyally if needed,... but I can't figure out why an software installer should be wiping-out a full virtual disk of data (even if it's data managed by that software).

**Comment by andrea-reale** [comment]
Hi @xmontero,

I am sorry about your issues. 
It is expected not to have a `docker-desktop-data` distribution anymore as there is no longer need of a separate distribution to hold your data. Your data is now stored in `<$LOCALAPPDATA>/wsl/disk/docker_data.vhdx`. This is not something introduced with the dev build. The expected behaviour is what you saw in your step 5 above.
What is not expected is to lose data after a reboot, of course.

Could you please share a diagnostics bundle and ID, so we can look more to understand what may have happened?

**Comment by neonima1995** [comment]
@theswissguy @xmontero @andrea-reale @WesWedding 
Friends, are you kidding? There is no need to delete data or reinstall anything there. There is a conflict for those who have more than one WSL Linux system installed. docker-desktop is also a Linux distribution, so Ubuntu and docker-desktop conflict with each other. Just need to configure in Docker or uninstall Ubuntu distribution!

**Comment by xmontero** [comment]
> Could you please share a diagnostics bundle and ID, so we can look more to understand what may have happened?

Sure! I'll provide everything needed!

Any link or pointer on how to get this data? I'm ready to read already-written articles if needed.

**Comment by xmontero** [comment]
Up to now, from a gitbash in Windows:

    xavi@msi-laptop:/c/usr_local/bin/ls all$ find /c/Users/xavi/AppData -iname "*.vhdx"
    /c/Users/xavi/AppData/Local/Docker/wsl/disk/docker_data.vhdx
    /c/Users/xavi/AppData/Local/Docker/wsl/main/ext4.vhdx
    find: ‘/c/Users/xavi/AppData/Local/ElevatedDiagnostics’: Permission denied
    find: ‘/c/Users/xavi/AppData/Local/Microsoft/Windows/INetCache/Low/Content.IE5’: Permission denied
    /c/Users/xavi/AppData/Local/Packages/CanonicalGroupLimited.Ubuntu20.04onWindows_79rhkp1fndgsc/LocalState/ext4.vhdx
    /c/Users/xavi/AppData/Local/Temp/4AC7A3AE-651D-487E-B4DD-07FE052498A2/swap.vhdx
    /c/Users/xavi/AppData/Local/Temp/65D2432F-AE03-44BC-BB35-72407D5C52B6/swap.vhdx
    find: ‘/c/Users/xavi/AppData/Local/Temp/msdtadmin’: Permission denied

So it seems that `/c/Users/xavi/AppData/Local/Docker/wsl/disk/docker_data.vhdx` exists.

I would now understand if it contains the "old data" (I hope!) or it's a brand new empty `docker_data` created by the installer. Should I try to "mount it manually" to inspect? Or better I don't touch anything?

**Comment by xmontero** [comment]
> Any link or pointer on how to get this data? I'm ready to read already-written articles if needed.

I'm going to carefully read this https://docs.docker.com/desktop/troubleshoot-and-support/troubleshoot/

If it's not what you need please tell me. Otherwise I'll try to "manually run the diagnose tool" (I'm not confident on launching the docker desktop again, I want to minimize chances of smashing any data).

**Comment by xmontero** [comment]
> Could you please share a diagnostics bundle and ID, so we can look more to understand what may have happened?

Here it is:
Diagnostics ID:     BB5FCCBA-A57A-422E-A449-A09F8402C9FD/20241202115602 (uploaded)

Is this what you need? Do you need anything else?

**Comment by andrea-reale** [comment]
@xmontero thanks for uploading the diagnostics. I went through the logs and I did not find anything suspicious. Docker Desktop does not have any code (except the factory reset and manual uninstall of course) that deletes your data, so I keep being confident your data is not lost.

Because this is getting a bit off topic for the issue at hand (the `more than one sd* disk in /sys/block` error - which appears so far to be mitigated by the [dev build](https://github.com/docker/for-win/issues/14453#issuecomment-2507992110)), would you mind opening another issue so I can try to provide more help for you there tomorrow, please?

**Comment by xmontero** [comment]
> would you mind opening another issue so I can try to provide more help for you there tomorrow, please?

Created "Lost data after a docker upgrade" #14461

**Comment by RuggMatt** [comment]
I'm getting the same error:

Diagnostics ID:
EB6FEC5B-2454-48F2-BE6A-B735489D72A4/20241202222120

I can usually get docker to start by doing `wsl --shutdwwn` and then restarting Docker Desktop, and everything is fine. 

Version 4.36.0

**Comment by maxippacheco** [comment]
> [@chris-riddell35](https://github.com/chris-riddell35), all: as promised here's a **development build** of Docker Desktop 4.36 that should **includes a fix for the issue**: [installer for windows x86](https://desktop-stage.docker.com/win/main/amd64/176956/Docker%20Desktop%20Installer.exe).
> 
> Please, let me know if that helps with the `more than one sd* disk` error discussed in this ticket, and feel free to provide any further feedback (make sure to share diagnostics in case you stumble into different errors).
> 
> **Note** that being a development build, the installer is unsigned and you'll need to tell Windows you trust it, before it will let you run it. You can do so by right-clicking on the file -> `Properties` -> `Unblock`.

It worked for me, you literally saved me LOL

**Comment by ginocampra** [comment]
Thanks guys, it worked great for me too

**Comment by drewtech** [comment]
> I'm getting the same error:
> 
> Diagnostics ID: EB6FEC5B-2454-48F2-BE6A-B735489D72A4/20241202222120
> 
> I can usually get docker to start by doing `wsl --shutdwwn` and then restarting Docker Desktop, and everything is fine.
> 
> Version 4.36.0

running `wsl --shutdown` and starting docker worked for me too - thanks!

**Comment by ffes** [comment]
Same problem when I already have an active WSL session. When I close that session, run `wsl --shutdown` Docker Desktop works as expected.

**Comment by BobDankert** [comment]
I also had this issue. Running `wsl --shutdown` first, then starting docker, then starting other wsl images (Ubuntu - started by just running `wsl`) after docker was running worked for me. I didn't try the dev build, as I'd rather wait for a new update to be published. Verified the state of WSL images using `wsl --list -v` as suggested above.

**Comment by CCadere** [comment]
> [@chris-riddell35](https://github.com/chris-riddell35), all: as promised here's a **development build** of Docker Desktop 4.36 that should **includes a fix for the issue**: [installer for windows x86](https://desktop-stage.docker.com/win/main/amd64/176956/Docker%20Desktop%20Installer.exe).
> 
> Please, let me know if that helps with the `more than one sd* disk` error discussed in this ticket, and feel free to provide any further feedback (make sure to share diagnostics in case you stumble into different errors).
> 
> **Note** that being a development build, the installer is unsigned and you'll need to tell Windows you trust it, before it will let you run it. You can do so by right-clicking on the file -> `Properties` -> `Unblock`.

OMG, thank you for this! Back to work now.

**Comment by qurm** [comment]
I had the above "more than one.." issue today, after allowing Docker Desktop to update itself to 4.36.0.   I am running Win 10.  Based on a comment above, I restarted and updated WSL as below.
After doing this, Docker Desktop started up fine, with no errors.  I did not need to rollback.
```
PS C:\WINDOWS\system32> wsl --update
Checking for updates.
Updating Windows Subsystem for Linux to version: 2.3.26.
PS C:\WINDOWS\system32> wsl --list
Windows Subsystem for Linux Distributions:
Ubuntu (Default)
docker-desktop-data
PS C:\WINDOWS\system32>
```

**Comment by CoolGoose** [comment]
For what is worth as an additional point of data

```running engine: waiting for the Docker API: engine linux/wsl failed to run: starting WSL engine: error spotted in wslbootstrap log: "[2024-12-05T09:09:40.255962887Z][wsl-bootstrap][F] more than one sd* disk in /sys/block with wwid ending by 57de6aa93848de02790adcb4: /dev/sdc,/dev/sde"```

```
 wsl --list
Windows Subsystem for Linux Distributions:
Ubuntu (Default)
docker-desktop
```

Diagnostics `ED24435D-FEB6-4E9A-9A23-585494CE6CEE/20241205090946`

Going to try the preview fix exe.

LE: seems that works indeed.

**Comment by JensKnipper** [comment]
The new build works for me, thanks!

**Comment by matrumz** [comment]
> Hello everyone,
> 
> thanks for reporting the problem. Could you please share the output of `wsl --list`?
> 
> From the diagnostics kindly shared by [@chris-riddell35](https://github.com/chris-riddell35) , it looks like WSL2 is in a state where it believes there are two disks with the same identifiers, which makes Docker Desktop boot fail.
> 
> Could you please confirm if shutting down WSL after quitting Docker Desktop (`wsl.exe --shutdown`) helps fixing the issue or not? (Alternative a Windows restart should also be equivalent)

@andrea-reale 

I can confirm after using this workaround several times that it works:
1. Close Docker Desktop
2. Close all WSL terminals, VSCode windows connected to WSL, etc.
3. In Powershell: `wsl --shutdown`
4. Open Docker Desktop (BEFORE STARTING WSL)

**Comment by jlegue** [comment]
I had a very similar issue in that Docker would not start on my Windows 10 system. I updated wsl2 as was described (wsl --update) which made things work with the previous build but the released 4.36.0 just would not start at all. Deployed the dev build on this ticket and was able to get Docker Desktop working once more. Thanks for the fix!

**Comment by RJFoster** [comment]
What is the likely timeframe for an official (non-dev) fix?

**Comment by benitogf** [comment]
> I had the same issue, but completely uninstalling and reinstalling resolved it for me.

this worked for me

**Comment by thangnq1001** [comment]
> I had the same issue. I solved it by going to 'Troubleshoot,' then clicking on 'Clean/Purge Data,' and selecting 'Hyper-V & WSL2.
> 
> ![Image](https://github.com/user-attachments/assets/3a46afcf-b0ef-48a3-aafa-8fcb5e75a056)
> 
> after that Docker Engine starts again.

Worked for me, no errors so far. I don't have any valuable data.

The popup may be stuck spinning, if that, you can click the background (to close the popup) and wait for the `Restart` button to clickable, then restart Docker.
![Image](https://github.com/user-attachments/assets/197ed857-0f55-4c33-8ecb-5bc8ccd1f898)

**Comment by lorenrh** [comment]
Docker Desktop 4.37.0 has just been released, and it contains a fix for this issue! see: https://docs.docker.com/desktop/release-notes/

if the problem persists, feel free to open a new issue!

**Comment by j-fbriere** [comment]
For me, no need to downgrade/upgrade/purge any data.
Running `wsl --shutdown` then starting Docker Desktop works for me. After that I can also use my other wsl distributions.

**Comment by NotVeryAsh** [comment]
> wsl.exe --update

Running only this command seemed to work for me 😄

**Comment by vivirox** [comment]
Still there for me. Two systems, one was/is experiencing after latest upgrade. The other system, completely fresh Windows install straight to 4.37.1. Windows 11. 

Neither Docker Desktop will ever actually start. The service, if on, gets stopped and disabled almost immediately. If 'Automatic' was set, that status gets washed. Never starts though.

**Comment by WesWedding** [comment]
To be clear, you're getting the error shown in the OP?

**Comment by jlegue** [comment]
Try doing a wsl --shutdown and then restarting Docker Desktop. It might
help.

**Comment by vivirox** [comment]
Yes. Same error. That's what brought me here.

I've tried about every possible 'workaround' or alleged fix possible. No dice.

**Comment by WesWedding** [comment]
What is your `wsl --list` output?

**Comment by vivirox** [comment]
wsl --list
Windows Subsystem for Linux Distributions:
Ubuntu (Default)
docker-desktop

**Comment by WesWedding** [comment]
Not sure what the appropriate way to get attention back into this issue is.  A full diagnostic report like the OP generated would probably help significantly, or perhaps there's a process to generate a support request.

**Comment by ksaber** [comment]
Guys, first time using Docker and for almost 2 weeks now i have tried almost every solution. If anyone can help my diagnostic code is "CDF346CC-88C2-4F95-84DF-33986A062672/20241227054541"

**Comment by edisanch** [comment]
I've been using docker for about a month on my windows 11 machine. Usinw WLS2. Currently Docker desktop is V4.37.1, yesterday, it crashed, 

running engine: waiting for the Docker API: engine linux/wsl failed to run: starting WSL engine: error spotted in wslbootstrap log: "[2025-01-30T10:22:22.849980818Z][wsl-bootstrap][F] exit status 32" I have tried everything in this issues thread, except basically removing my wsl. As I have lots of work there that would take time for me to backup and reproduce. 
Anyone has any idea ?

Diagnostics ID. A54CC16C-3CF3-417A-8F05-A436658473EE/20250130101631

**Comment by WesWedding** [comment]
I would recommend finding a way to open a different issue or trying to find a different thread, this one doesn't obviously match the specific issue you are encountering without additional log information.  

Consider opening a ticket here: https://hub.docker.com/support/contact/

**Comment by lamy** [comment]
> ### Description
> 
> Unable to start docker desktop version 4.36.0; I am consistently seeing the following error:
> 
> `running engine: waiting for the Docker API: engine linux/wsl failed to run: starting WSL engine: error spotted in wslbootstrap log: "[2024-11-28T16:04:44.799442528Z][wsl-bootstrap][F] more than one sd* disk in /sys/block with wwid ending by 57de6aa93848de02790adcb4: /dev/sdb,/dev/sdd"`
> 
> 
> 
> ### Reproduce
> 
> Start docker desktop
> 
> ### Expected behavior
> 
> Docker desktop should start successfully
> 
> ### docker version
> 
> ```bash
> Client:
>  Version:           27.3.1
>  API version:       1.47
>  Go version:        go1.22.7
>  Git commit:        ce12230
>  Built:             Fri Sep 20 11:42:27 2024
>  OS/Arch:           windows/amd64
>  Context:           desktop-linux
> ```
> 
> ### docker info
> 
> ```bash
> Client:
>  Version:    27.3.1
>  Context:    desktop-linux
>  Debug Mode: false
>  Plugins:
>   ai: Ask Gordon - Docker Agent (Docker Inc.)
>     Version:  v0.1.0
>     Path:     C:\Program Files\Docker\cli-plugins\docker-ai.exe
>   buildx: Docker Buildx (Docker Inc.)
>     Version:  v0.18.0-desktop.2
>     Path:     C:\Program Files\Docker\cli-plugins\docker-buildx.exe
>   compose: Docker Compose (Docker Inc.)
>     Version:  v2.30.3-desktop.1
>     Path:     C:\Program Files\Docker\cli-plugins\docker-compose.exe
>   debug: Get a shell into any image or container (Docker Inc.)
>     Version:  0.0.37
>     Path:     C:\Program Files\Docker\cli-plugins\docker-debug.exe
>   desktop: Docker Desktop commands (Alpha) (Docker Inc.)
>     Version:  v0.0.15
>     Path:     C:\Program Files\Docker\cli-plugins\docker-desktop.exe
>   dev: Docker Dev Environments (Docker Inc.)
>     Version:  v0.1.2
>     Path:     C:\Program Files\Docker\cli-plugins\docker-dev.exe
>   extension: Manages Docker extensions (Docker Inc.)
>     Version:  v0.2.27
>     Path:     C:\Program Files\Docker\cli-plugins\docker-extension.exe
>   feedback: Provide feedback, right in your terminal! (Docker Inc.)
>     Version:  v1.0.5
>     Path:     C:\Program Files\Docker\cli-plugins\docker-feedback.exe
>   init: Creates Docker-related starter files for your project (Docker Inc.)
>     Version:  v1.4.0
>     Path:     C:\Program Files\Docker\cli-plugins\docker-init.exe
>   sbom: View the packaged-based Software Bill Of Materials (SBOM) for an image (Anchore Inc.)
>     Version:  0.6.0
>     Path:     C:\Program Files\Docker\cli-plugins\docker-sbom.exe
>   scout: Docker Scout (Docker Inc.)
>     Version:  v1.15.0
>     Path:     C:\Program Files\Docker\cli-plugins\docker-scout.exe
> 
> Server:
> ERROR: Get "http://%2F%2F.%2Fpipe%2FdockerDesktopLinuxEngine/v1.47/info": context canceled
> errors pretty printing info
> ```
> 
> ### Diagnostics ID
> 
> 1AA58D3A-8AF2-493C-AEAF-41A589D2DF25/20241128162112
> 
> ### Additional Info
> 
> WSL version info:
> ```
> wsl --version
> WSL version: 2.3.26.0
> Kernel version: 5.15.167.4-1
> WSLg version: 1.0.65
> MSRDC version: 1.2.5620
> Direct3D version: 1.611.1-81528511
> DXCore version: 10.0.26100.1-240331-1435.ge-release
> Windows version: 10.0.22631.4460
> ```

**Comment by lamy** [comment]
lamy/github

**Comment by Winand** [comment]
FINALLY I've managed to fix this issue on 4.36.0 by changing the VHDX disk identifier: `Set-VHD -Path "$env:USERPROFILE\AppData\Local\Docker\wsl\disk\docker_data.vhdx" -ResetDiskIdentifier`

First I checked the ids of /dev/sd* devices via docker-desktop WSL distro without trying to start Docker: `for disk in /sys/block/sd*; do echo -n "$disk: "; cat "$disk/device/wwid" 2>/dev/null; done`

Then I checked ID of the Docker's data disk: `Get-VHD -Path "$env:USERPROFILE\AppData\Local\Docker\wsl\disk\docker_data.vhdx"`

There's already a /dev/sd* device with the same ID as my VHDX file. When I try to start Docker Desktop it tries to add a device with this ID again what leads to a confict.
